tests/ragger/boilerplate_client/boilerplate_cmd_builder.py

```diff
@@ -0,0 +1,218 @@
+
+        return header + cdata
+
+    def get_app_and_version(self) -> bytes:
+        """Command builder for GET_APP_AND_VERSION (builtin in BOLOS SDK).
+
+        Returns
+        -------
+        bytes
+            APDU command for GET_APP_AND_VERSION.
+
+        """
+        return self.serialize(cla=0xB0,  # specific CLA for BOLOS
+                              ins=0x01,
+                              p1=0x00,
+                              p2=0x00,
+                              cdata=b"")
+
+    def get_version(self) -> bytes:
+        """Command builder for GET_VERSION.
+
+        Returns
+        -------
+        bytes
+            APDU command for GET_VERSION.
+
+        """
+        return self.serialize(cla=self.CLA,
+                              ins=InsType.INS_GET_VERSION,
+                              p1=0x00,
+                              p2=0x00,
+                              cdata=b"")
+
+    def get_app_name(self) -> bytes:
+        """Command builder for GET_APP_NAME.
+
+        Returns
+        -------
+        bytes
+            APDU command for GET_APP_NAME.
+
+        """
+        return self.serialize(cla=self.CLA,
+                              ins=InsType.INS_GET_APP_NAME,
+                              p1=0x00,
+                              p2=0x00,
+                              cdata=b"")
+
+    def get_public_key(self, bip32_path: str, display: bool = False) -> bytes:
+        """Command builder for GET_PUBLIC_KEY.
+
+        Parameters
+        ----------
+        bip32_path: str
+            String representation of BIP32 path.
+        display : bool
+            Whether you want to display the address on the device.
+
+        Returns
+        -------
+        bytes
+            APDU command for GET_PUBLIC_KEY.
+
+        """
+        bip32_paths: List[bytes] = bip32_path_from_string(bip32_path)
+
+        cdata: bytes = b"".join([
+            len(bip32_paths).to_bytes(1, byteorder="big"),
+            *bip32_paths
+        ])
+
+        return self.serialize(cla=self.CLA,
+                              ins=InsType.INS_GET_PUBLIC_KEY,
+                              p1=0x01 if display else 0x00,
+                              p2=0x00,
+                              cdata=cdata)
+
+    def sign_tx(self, bip32_path: str, transaction: Transaction) -> Iterator[Tuple[bool, bytes]]:
+        """Command builder for INS_SIGN_TX.
+
+        Parameters
+        ----------
+        bip32_path : str
+            String representation of BIP32 path.
+        transaction : Transaction
+            Representation of the transaction to be signed.
+
+        Yields
+        -------
+        bytes
+            APDU command chunk for INS_SIGN_TX.
+
+        """
+        bip32_paths: List[bytes] = bip32_path_from_string(bip32_path)
+
+        cdata: bytes = b"".join([
+            len(bip32_paths).to_bytes(1, byteorder="big"),
+            *bip32_paths
+        ])
+
+        yield False, self.serialize(cla=self.CLA,
```

Comment:
What I made for a similar use case in Solana
```
    def send_first_message_batch(self, messages: List[bytes], p1: int) -> RAPDU:
        self._client.exchange(CLA, INS.INS_SIGN_MESSAGE, p1, P2_MORE, messages[0])
        for m in messages[:1]:
            self._client.exchange(CLA, INS.INS_SIGN_MESSAGE, p1, P2_MORE | P2_EXTEND, m)


    @contextmanager
    def send_async_sign_message(self,
                                derivation_path : bytes,
                                message: bytes) -> Generator[None, None, None]:
        message_splited_prefixed = self.split_and_prefix_message(derivation_path, message)

        # Send all chunks with P2_MORE except for the last chunk
        # Send all chunks with P2_EXTEND except for the first chunk
        if len(message_splited_prefixed) > 1:
            final_p2 = P2_EXTEND
            self.send_first_message_batch(message_splited_prefixed[:-1], P1_CONFIRM)
        else:
            final_p2 = 0

        with self._client.exchange_async(CLA,
                                         INS.INS_SIGN_MESSAGE,
                                         P1_CONFIRM,
                                         final_p2,
                                         message_splited_prefixed[-1]):
            yield
```

Reply:
Not sure why we yield while sending P2_NONCONFIRM data ? Shouldn't we keep sending until the last P2_CONFIRM message ?

Reply:
To be cleaned in another PR